stream/index.ts

```diff
@@ -174,16 +174,18 @@ export async function verifyAndIndexStream(
   return items;
 }
 
-async function hasEnough(
+async function readBytes(
   reader: AsyncGenerator<Buffer>,
   buffer: Uint8Array,
   length: number,
 ): Promise<Uint8Array> {
   if (buffer.byteLength > length) return buffer;
-  buffer = Buffer.concat([buffer, (await reader.next()).value]);
-  if (buffer.byteLength > length) return buffer;
 
-  return buffer;
+  const { done, value } = await reader.next();
+
+  if (done && buffer.byteLength < length) throw new Error("Invalid buffer");
+
+  return readBytes(reader, Buffer.concat([buffer, value]), length);
```

Comment:
Correct me if I'm missing out the logic here, but wont this result in recursion until error is thrown? It wasn't recursive before?

The name is good, apologies for the delay, just need to understand :)